mobly/suite_runner.py

```diff
@@ -70,7 +107,89 @@ def run_suite(test_classes, argv=None):
       help='A list of test classes and optional tests to execute.')
   if not argv:
     argv = sys.argv[1:]
-  args = parser.parse_args(argv)
+  return parser.parse_args(argv)
+
+
+def _find_suite_class():
+  """Finds the test suite class in the current module.
+
+  Walk through module members and find the subclass of BaseSuite. Only
+  one subclass is allowed in a module.
+
+  Returns:
+      The test suite class in the test module.
+  """
+  test_suites = []
+  main_module_members = sys.modules['__main__']
+  for _, module_member in main_module_members.__dict__.items():
+    if inspect.isclass(module_member):
+      if issubclass(module_member, base_suite.BaseSuite):
+        test_suites.append(module_member)
+  if len(test_suites) != 1:
+    logging.error('Expected 1 test class per file, found %s.',
+                  [t.__name__ for t in test_suites])
+    sys.exit(1)
+  return test_suites[0]
+
+
+class _SuiteRunner(object):
+  """Class used to execute a Mobly suite in google3."""
+
+  def __init__(self, suite, args):
+    cli_args = _parse_cli_args(args)
+    test_configs = config_parser.load_test_config_file(cli_args.config)
+    config_count = len(test_configs)
+    if config_count != 1:
+      logging.error('Expect exactly one test config, found %d', config_count)
+    self._config = test_configs[0]
+    self._runner = test_runner.TestRunner(
+        log_dir=self._config.log_path, testbed_name=self._config.testbed_name)
+    self._suite = suite(self._runner, self._config)
+
+  def run(self):
+    """Runs the test suite.
+
+    This should not be called by users directly.
+    """
+    ok = False
+    with self._runner.mobly_logger():
+      try:
+        self._suite.setup_suite(self._config.copy())
+        try:
+          self._runner.run()
+          ok = self._runner.results.is_all_pass
+          print(ok)
+        except signals.TestAbortAll:
+          pass
+      finally:
+        self._suite.teardown_suite()
+    if not ok:
+      sys.exit(1)
+
+
+def run_suite_class(argv=None):
```

Comment:
let's define an order for members of a module?
usually functions come before classes, except error types